### PR TITLE
Make vet docs be per note.

### PR DESCRIPTION
**Pull request by bmarshall13**
Replace the vetter docs with docs that are per-note. The content came from outside this repo, and still need to be
updated for newer istio versions.

### CODE REVIEW

File: pkg/vetter/serviceportprefix/README-missing-service-port-prefix.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+rename the service port to `http2-client`.
+
+If the service port is using a layer 7 protocol that the sidecar does not
+understand, you should name the service port with a `tcp-` or `udp-` in front
+of it to indicate that you are opting out of layer 7 policy from the service
+mesh for this service port.  For instance, if your service port named `backend`
+is using an unknown protocol that runs on top of tcp, rename the service port
+to `tcp-backend`.
+
+In version 0.2.12, these protocols are supported: `grpc`, `https`, `http2`,
```
Comment (nrjpoddar) [inline pkg/vetter/serviceportprefix/README-missing-service-port-prefix.md]: In version 0.2.12 -> In version 0.8?

File: pkg/vetter/mtlsprobes/README-mtls-probes-incompatible.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+- **Disable liveness probes.** You can remove liveness and readiness probes
+  from your pod spec.  In this case, the kubelet will not be able to determine
+when your pod is ready to enter service, and if it has become unhealthy.  So,
+as long as your pod is running, it will be considered healthy.  If you choose
+this technique, make sure your pod will exit if it becomes unhealthy.
+
+- **Upgrade Istio.** This issue will be fixed in a future version of istio but
+  is not yet fixed in any released version.
+
+## See also
```
Comment (nrjpoddar) [inline pkg/vetter/mtlsprobes/README-mtls-probes-incompatible.md]: Should we remove ## See also section from these READMEs.

File: pkg/vetter/serviceportprefix/README-missing-service-port-prefix.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+`http`, `tcp`, `udp`, `mongo`, `redis`.
+
+## See Also
```
Comment (nrjpoddar) [inline pkg/vetter/serviceportprefix/README-missing-service-port-prefix.md]: Should we remove ## See Also?

File: pkg/vetter/serviceportprefix/README-missing-service-port-prefix.md (hunk 1 above)
Comment (bmarshall13) [reply pkg/vetter/serviceportprefix/README-missing-service-port-prefix.md]: I want to restructure the content as is, then update it.

**Pull request merged**